providers/nodesmith-provider.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+var __extends = (this && this.__extends) || (function () {
+    var extendStatics = Object.setPrototypeOf ||
+        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
+        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
+    return function (d, b) {
+        extendStatics(d, b);
+        function __() { this.constructor = d; }
+        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
+    };
+})();
+var __importStar = (this && this.__importStar) || function (mod) {
+    if (mod && mod.__esModule) return mod;
+    var result = {};
+    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
+    result["default"] = mod;
+    return result;
+};
+Object.defineProperty(exports, "__esModule", { value: true });
+var json_rpc_provider_1 = require("./json-rpc-provider");
+var networks_1 = require("../utils/networks");
+var properties_1 = require("../utils/properties");
+var errors = __importStar(require("../errors"));
+// Provider for connecting to Nodesmith's hosted JSON RPC endpoints
+var NodesmithProvider = /** @class */ (function (_super) {
+    __extends(NodesmithProvider, _super);
+    function NodesmithProvider(apiKey, network) {
```

Comment:
Yes, regardless hows it would be accepted. Network should be a first param, and apiKey should be the second, as to have only one provider with reversed params is really adds misconception.